### PR TITLE
LSP ProjectView test + fixes

### PR DESCRIPTION
**Pull request by sdedic**
This PR focus on integration-testing the projects view. It checks basic scenarios: created files appear, deleted disappear and the package nodes restructure according to the occupied folders.

I moved certain registrations / layer folders from `nbcode` integration module to LSP module - these are harmless for IDE execution, but the LSP server (which can be potentially started inside NB IDE) does not function without those folders well. Also `foundProjects` view is defined in the LSP server but its critical cookie registrations were in the integration module - a different distribution.

During testing, I've encountered a NPE thrown when the client asks for details of a non-existing node: this is fixed by the last commit.

### CODE REVIEW

File: java/java.lsp.server/test/unit/data/gradle-hello/app/gradlew [hunk 1]
```diff
@@ -0,0 +1,234 @@
+#!/bin/sh
```
Comment (JaroslavTulach) [inline java/java.lsp.server/test/unit/data/gradle-hello/app/gradlew:1]: Ugly file, too ugly. Would it be to dare to ask you to create a Gradle project using the wizard/template in the IDE?

**Review by JaroslavTulach — APPROVED** [review]
It is sad that two biggest files in the change are `gradlew` wrappers! I'd prefer to generate them using "gradle init" which is available when one uses "New From Template" on the gradle project. I know you argued that JAR files should rather be in the repository and this is similar issue, so I leave the decision up to you.

File: java/java.lsp.server/test/unit/data/gradle-hello/app/gradlew (hunk 1 above)
Comment (sdedic) [reply java/java.lsp.server/test/unit/data/gradle-hello/app/gradlew:1]: ok, why not.

Comment (sdedic) [reply java/java.lsp.server/test/unit/data/gradle-hello/app/gradlew]: project template used in 7cce3cf

**Review by dbalek — APPROVED** [review]
Looks fine.

**Pull request merged**